test/libsolidity/util/TestFunctionCall.cpp

```diff
@@ -266,6 +266,7 @@ string TestFunctionCall::formatBytesParameters(
 			if (preferredParams)
 			{
 				ContractABIUtils::overwriteParameters(_errorReporter, preferredParams.value(), abiParams.value());
+				preferredParams.value().begin()->abiType = abiParams.value().begin()->abiType;
```

Comment:
Can you explain why this was introduced? Is it needed?

Reply:
`abiParams` refer to the actual parameter information that is defined by the contract ABI, where `preferredParams` is basically `_parameters` that holds the parameter information that was actually supplied in test case file.

E.g. the `abiType` of the result type of `// u8x2(ufixed8x2): 2.55 -> 2.552` would be `{type=UnsignedFixedPoint, fractionalDigits=2}` in `abiParams`, because it is defined by the contract ABI, `function u8x2(ufixed8x2 i) public pure returns (ufixed8x2 x)`. (`ufixed8x2` -> `fractionalDigits=2`), where `preferredParams`/`_parameter` will point to the parsed result of `2.552` that will have `fractionalDigits=3`. 

For the formatting we want to use the type that is defined by the contract ABI. Thats why I just overwrite the `abiType` here with the type defined by the contract ABI.

While I'm writing this, I think this could be improved by adding this to `ContractABIUtils::overwriteParameters`.

Reply:
Ok, I just changed `ContractABIUtils::overwriteParameters` so that it will also overwrite the parameters, if the `abiType.fractionalDigits` are different.